Implement Notifications for Updates from Saved Startups #45 v1

Nazar-j opened the pull request.

mehalyna commented on notifications/signals.py:
Both signal handlers create a `Notification` and send an email. Consider moving that logic into a reusable helper function to avoid code repetition.

mehalyna commented on notifications/tests/test_triger_update_profile.py:
_Optional:_ Parametrize to test multiple followers. This helps confirm the signal loops through all relevant followers:
```
@pytest.mark.parametrize("num_followers", [1, 3])
def test_notifications_sent_to_all_followers(num_followers):
    ...
    for i in range(num_followers):
        user = User.objects.create_user(email=f"investor{i}@example.com", password="testpass")
        investor = CompanyProfile.objects.create(company_name=f"Investor {i}", type="enterprise")
        UserToCompany.objects.create(user=user, company=investor)
        CompanyFollowers.objects.create(investor=investor, startup=startup_profile)
    
    startup_profile.company_name = "Startup X - Updated"
    startup_profile.save()

    assert Notification.objects.count() == num_followers
```

Nazar-j commented:
In the last commit, I updated signals.py and fixed the project failure: I had committed a non-existent serializer and restored the view that Roman removed in commit 0cec43f (Resolved merge conflict in views.py).
